drivers/sensor/ti_hdc2xxx/ti_hdc2xxx.c

```diff
@@ -0,0 +1,165 @@
+	uint16_t buf[2];
+	int rc;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	/* start conversion of both temperature and humidity with the default accuracy (14 bits) */
+	rc = i2c_reg_write_byte(drv_cfg->bus, drv_cfg->i2c_addr, TI_HDC2XXX_REG_MEAS_CFG, 0x01);
+	if (rc < 0) {
+		LOG_DBG("Failed to write measurement configuration register");
+		return rc;
+	}
+
+	/* for 14-bit conversion temperature needs 660us and humidity 610us */
+	k_msleep(2);
+
+	/* temperature and humidity registers are consecutive, read them in the same burst */
+	rc = i2c_burst_read(drv_cfg->bus, drv_cfg->i2c_addr,
+			    TI_HDC2XXX_REG_TEMP, (uint8_t *)buf, sizeof(buf));
+	if (rc < 0) {
+		LOG_DBG("Failed to read sample data");
+		return rc;
+	}
+
+	drv_data->t_sample = sys_le16_to_cpu(buf[0]);
+	drv_data->rh_sample = sys_le16_to_cpu(buf[1]);
+
+	return 0;
+}
+
+
+static int ti_hdc2xxx_channel_get(const struct device *dev,
+				  enum sensor_channel chan,
+				  struct sensor_value *val)
+{
+	struct ti_hdc2xxx_data *drv_data = dev->data;
+	int32_t tmp;
+
+	/* See datasheet "Register Maps" section for more details on processing sample data. */
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		/* val = -40 + 165 * sample / 2^16 */
+		tmp = drv_data->t_sample * 165U;
+		val->val1 = (tmp >> 16) - 40;
+		/* x * 1000000 / 65536 = x * 15625 / 1024 */
+		val->val2 = ((tmp & 0xFFFF) * 15625U) >> 10;
+		break;
+	case SENSOR_CHAN_HUMIDITY:
+		/* val = 100 * sample / 2^16 */
+		tmp = drv_data->rh_sample * 100U;
+		val->val1 = tmp >> 16;
+		/* x * 1000000 / 65536 = x * 15625 / 1024 */
+		val->val2 = ((tmp & 0xFFFF) * 15625U) >> 10;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api ti_hdc2xxx_api_funcs = {
+	.sample_fetch = ti_hdc2xxx_sample_fetch,
+	.channel_get = ti_hdc2xxx_channel_get,
+};
+
+static int ti_hdc2xxx_init(const struct device *dev)
+{
+	const struct ti_hdc2xxx_config *drv_cfg = dev->config;
+	uint16_t buf[2];
+	int rc;
+
+	if (!device_is_ready(drv_cfg->bus)) {
+		LOG_DBG("I2C bus %s not ready", drv_cfg->bus->name);
+		return -ENODEV;
+	}
+
+	/* manufacturer and device ID registers are consecutive, read them in the same burst */
+	rc = i2c_burst_read(drv_cfg->bus, drv_cfg->i2c_addr,
+			    TI_HDC2XXX_REG_MANUFACTURER_ID, (uint8_t *)buf, sizeof(buf));
+	if (rc < 0) {
+		LOG_DBG("Failed to read manufacturer and device IDs");
+		return rc;
+	}
+
+	if (sys_le16_to_cpu(buf[0]) != TI_HDC2XXX_MANUFACTURER_ID) {
+		LOG_ERR("Failed to get correct manufacturer ID");
+		return -EINVAL;
+	}
+	if (sys_le16_to_cpu(buf[1]) != TI_HDC2XXX_DEVICE_ID) {
+		LOG_ERR("Unsupported device ID");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+/* Initializes a struct ti_hdc2xxx_config for an instance on a I2C bus. */
+#define TI_HDC2XXX_CONFIG_I2C(inst)						\
+	{								\
+		.bus = DEVICE_DT_GET(DT_INST_BUS(inst)),		\
+		.tf = &ti_hdc2xxx_i2c_transfer_function,			\
+		.bus_cfg.i2c_addr = DT_INST_REG_ADDR(inst)		\
+	}
```

Comment:
This definition does not seem to be used

Reply:
Correct, I have just removed it.